test/show.jl

```diff
@@ -755,6 +755,37 @@ end
 
 @test startswith(sprint(show, typeof(x->x), context = :module=>@__MODULE__), "var\"")
 
+# PR 53719
+module M53719
+    f = x -> x + 1
+    function foo(x)
+        function bar(y)
+            function baz(z)
+                return x + y + z
+            end
+            return baz
+        end
+        return bar
+    end
+    function foo2(x)
+        function bar2(y)
+            return z -> x + y + z
+        end
+        return bar2
+    end
+end
+
+# typeof(M53719.f) should be printed as var"#[0-9]+#[0-9]+"
+@test occursin(r"var\"#[0-9]+#[0-9]+", sprint(show, typeof(M53719.f), context = :module=>M53719))
+#typeof(M53719.foo(1)) should be printed as var"#bar#foo##[0-9]+"
+@test occursin(r"var\"#bar#foo##[0-9]+", sprint(show, typeof(M53719.foo(1)), context = :module=>M53719))
+#typeof(M53719.foo(1)(2)) should be printed as var"#baz#foo##[0-9]+"
+@test occursin(r"var\"#baz#foo##[0-9]+", sprint(show, typeof(M53719.foo(1)(2)), context = :module=>M53719))
+#typeof(M53719.foo2(1)) should be printed as var"#bar2#foo2##[0-9]+"
+@test occursin(r"var\"#bar2#foo2##[0-9]+", sprint(show, typeof(M53719.foo2(1)), context = :module=>M53719))
+#typeof(M53719.foo2(1)(2)) should be printed as var"#foo2##[0-9]+#foo2##[0-9]+"
+@test occursin(r"var\"#foo2##[0-9]+#foo2##[0-9]+", sprint(show, typeof(M53719.foo2(1)(2)), context = :module=>M53719))
```

Comment:
You currently only have tests for the `typeof(f)`, but your PR also makes adjustments to the printing for `f` itself. Can you please add test cases for those too?